src/connection.ts

```diff
@@ -1369,6 +1369,9 @@ class Connection extends EventEmitter {
       this.messageIo.on('data', (data) => { this.dispatchEvent('data', data); });
       this.messageIo.on('message', () => { this.dispatchEvent('message'); });
       this.messageIo.on('secure', (cleartext) => { this.emit('secure', cleartext); });
+      this.messageIo.on('error', (error) => {
+        this.emit('connect', error)
```

Comment:
@IanChokS I think there's still a problem with this. 🙇‍♂️ 

If a protocol error happens when establishing a connection, this will emit a `connect` event, but I don't think this will correctly close the network socket or perform any other cleanup.

On top of that, if a protocol error happens _after_ establishing a connection initially, this will fire a second, unexpected `connect` event, and most likely lead to really weird behavior in code that uses `tedious`.

I still think calling the `socketError` is probably the correct thing to do. It will emit a `connect` event with an error if the connection was not fully established yet, and will emit an `error` if the connection was established already. It will also dispatch the `socketError` internal event that will cause the connection to switch to the `FINAL` state, which will ensure the underlying network socket will get closed correctly.